orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/persistence/ExecutionRepositoryTck.groovy

```diff
@@ -516,12 +528,112 @@ class JedisExecutionRepositorySpec extends ExecutionRepositoryTck<JedisExecution
     retrieved.size() == actual
 
     where:
-    limit  || actual
-    0      || 0
-    1      || 1
-    2      || 2
-    4      || 4
-    100    || 4
+    limit || actual
+    0     || 0
+    1     || 1
+    2     || 2
+    4     || 4
+    100   || 4
+  }
+
+  def "can retrieve orchestrations from multiple redis stores"() {
+    given:
+    repository.store(new Orchestration(application: "orca", executingInstance: "current"))
+    repository.store(new Orchestration(application: "orca", executingInstance: "current"))
+    repository.store(new Orchestration(application: "orca", executingInstance: "current"))
+
+    and:
+    def previousRepository = new JedisExecutionRepository(new NoopRegistry(), jedisPoolPrevious, Optional.empty(), 1, 50)
+    previousRepository.store(new Orchestration(application: "orca", executingInstance: "previous"))
+    previousRepository.store(new Orchestration(application: "orca", executingInstance: "previous"))
+    previousRepository.store(new Orchestration(application: "orca", executingInstance: "previous"))
+
+    when:
+    // TODO-AJ limits are current applied to each backing redis
```

Comment:
You could maybe apply the limit to the merged one using `take(limit)` but I'm not sure if it would preserve ordering correctly or that would require more work